fvm/src/machine/default.rs

```diff
@@ -211,15 +211,9 @@ where
     fn create_actor(&mut self, addr: &Address, act: ActorState) -> Result<ActorID> {
         let state_tree = self.state_tree_mut();
 
-        let addr_id = state_tree
-            .register_new_address(addr)
-            .context("failed to register new address")
-            .or_fatal()?;
+        let addr_id = state_tree.register_new_address(addr)?;
 
-        state_tree
-            .set_actor(&Address::new_id(addr_id), act)
-            .context("failed to set actor")
-            .or_fatal()?;
+        state_tree.set_actor(addr_id, act)?;
```

Comment:
These now internally decide if the error should be fatal, and don't treat the error as fatal if, e.g., we're in "readonly" mode.